yt_dlp/extractor/bundestag.py

```diff
@@ -0,0 +1,127 @@
+    _SHARE_VIDEO_REGEX = r'/\d+_(?P<codec>\w+)_(?P<width>\w+)_(?P<height>\w+)_(?P<bitrate>\d+)kb_\w+_(?P<lang>\w+)_\d+\.(?P<ext>\w+)'
+
+    def _bt_extract_share_formats(self, video_id):
+        share_data = self._download_json(
+            f'{self._SHARE_URL}{video_id}', video_id, note='Downloading share format JSON')
+        error = traverse_obj(share_data, ('status', 'code', {int}))
+        if error != 1:
+            self.report_warning(format_field(
+                share_data, [('status', 'message', {str})],
+                'Share API response: %s', default='Unknown Share API Error')
+                + bug_reports_message())
+            return
+
+        for name, url in share_data.items():
+            if not isinstance(name, str) or not name.startswith('audio') or not url_or_none(url):
+                continue
+
+            match = re.search(self._SHARE_AUDIO_REGEX, url)
+            yield {
+                'format_id': name,
+                'url': url,
+                'vcodec': 'none',
+                **traverse_obj(match, {
+                    'acodec': 'codec',
+                    'abr': ('bitrate', {int_or_none}),
+                    'ext': 'ext',
+                }),
+            }
+
+        for name, url in share_data.items():
+            if not isinstance(name, str) or not name.startswith('download') or not url_or_none(url):
+                continue
+
+            match = re.search(self._SHARE_VIDEO_REGEX, url)
+            yield {
+                'format_id': name,
+                'url': url,
+                **traverse_obj(match, {
+                    'vcodec': 'codec',
+                    'tbr': ('bitrate', {int_or_none}),
+                    'width': ('width', {int_or_none}),
+                    'height': ('height', {int_or_none}),
+                    'ext': 'ext',
+                }),
+            }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        formats = []
+        result = {'id': video_id, 'formats': formats}
+
+        try:
+            formats.extend(self._extract_m3u8_formats(self._INSTANCE_FORMAT.format(video_id), video_id))
+        except ExtractorError as error:
+            if isinstance(error.cause, HTTPError) and error.cause.status == 404:
+                raise ExtractorError('Could not find video id', expected=True) from None
+            self.report_warning(f'Error extracting hls formats: {error}', video_id)
+        formats.extend(self._bt_extract_share_formats(video_id))
+        if not formats:
+            self.raise_no_formats('Could not find suitable formats', video_id=video_id)
+
+        metadata_overlay_html = self._download_webpage(
+            self._OVERLAY_URL, video_id,
+            query={'videoid': video_id, 'view': 'main'},
+            note='Downloading metadata overlay', fatal=False)
+        if isinstance(metadata_overlay_html, str) and not metadata_overlay_html.isspace():
```

Comment:
```suggestion
        metadata_overlay_html = self._download_webpage(
            self._OVERLAY_URL, video_id,
            query={'videoid': video_id, 'view': 'main'},
            note='Downloading metadata overlay', fatal=False) or ''
        if metadata_overlay_html.strip():
```

Reply:
why use `strip` over `isspace`?

Reply:
`not ''.isspace()` is truthy

Reply:
Thats exactly what we want. The overlay gives back only space if it is in an erroneous state. This state is different from when the page failed to download completely. Alternatively we can just not print a warning at all since it will result in core code telling us that a generic title had to be created. This would implicitly give the same info